pages/content/posts/localizing-your-website.md

```diff
@@ -0,0 +1,29 @@
+date: "2022-12-14"
+tags: ["development"]
+ShowToc: false
+ShowBreadCrumbs: false
+excerpt: "A client of mine had recently asked me why it was so important to localize their web app. They asked me, why spend money on localization when you can do it for free with Google Translate? There is so much to unpack about why that approach and mindset is wrong."
+---
+
+A client of mine had recently asked me why it was so important to localize their web app.  They told me: “we just plug Google Translate in and it’s been fine. Why spend hundreds, thousands, tens of thousands on localization when you can do it for free with Google Translate?”
+
+There is _so_ much to unpack about why that approach and mindset is _wrong_.
+
+There are many different categories to consider when it comes to inclusivity of diverse audiences, whether it’s to accommodate sensory impairments, theming (dark/light mode), localization and more.  When designing your website/software, or perhaps updating it, one should consider every aspect of inclusion every step of the way to engage as wide of an audience as possible.  It is important that no portion is overlooked either, as then you risk your product appearing broken or incomplete.
+
+Take alt. text attributes for images, for example.  Let’s say you have ten images, and you have properly and accurately described nine of the images using alt. text attributes (or marked them as decorative, but either way, you have followed accessibility standards for 90% of your images).  A visually impaired user that needs these attributes then goes to the page containing the image without an alt. text attribute, and suddenly, an entire element cannot be described to them.  Is it just decoration, or does it contain critical information that is essential to the rest of the page?  The screen reader doesn’t know (or maybe it could, but that’s a very big “maybe”), and the user doesn’t know.  To the visually impaired user, the web page would simply be broken.
+
+The same can be said for localization.  Now, in this case, there is already very powerful machine translation software (such as Google Translate) that is able to translate entire documents, webpages, and even verbal conversations, and they’re only going to get more powerful.  However, machine translations can only do so much: the ones that are readily available to the common folk cannot comprehend context, including user interface options or special phrases (product names, etc.).  Additionally, machine translation is not able to capture the nuances and subtleties of languages.  Often times, this results in the wrong word selection, translations of phrases that shouldn’t be translated, inconsistency in word selection throughout paragraphs, and even completely wrong translations.  (I ran into this when attempting to automatically translate a Russian website to English, and the word selection was so bad, I had to ask my Russian translator to translate it for me instead.)
```

Comment:
```suggestion
The same can be said for localization.  Now, in this case, there is already very powerful machine translation software (such as [Google Translate](https://translate.google.com/) that is capable of translating entire documents, webpages, and even oral conversations, and these translation capabilities are only going to get more powerful.  However, automated services can only do so much: the ones that are readily available provide translations only; they cannot understand context, including user interface options, idioms, or other special phrases such as product names, etc.  

Machine translations are not fully capable of capturing the nuances and subtleties of languages; they can translate but can't localize. Translations often result in the wrong word selection, translations of phrases that shouldn’t be translated (hot dog is "hot dog" everywhere; no one wants to eat a "_<span lang="fr-fr">chien chaud</span>_"), inconsistency in word selection throughout paragraphs, and even completely wrong translations.  
```

I removed the Russian translation part, because that is the point of this whole article, not a specific example for this section. I added the french hot dog instead. Feel free to replace my hot dog example with an example from the Russian translation.... that;s why I added it: provided a suggested template for the edit

Reply:
The same can be said for localization.  Now, in this case, there is already very powerful machine translation software (such as [Google Translate](https://translate.google.com/)) that is capable of translating entire documents, webpages, and even oral conversations, and these translation capabilities are only going to get more powerful.  However, automated services can only do so much: the ones that are readily available provide translations only; they cannot understand context, including user interface options, idioms, or other special phrases such as product names, etc.

Machine translations are not fully capable of capturing the nuances and subtleties of languages; they can translate but can't localize. Translations often result in the wrong word selection, translations of phrases that shouldn’t be translated (hot dog is "hot dog" everywhere; no one wants to eat a "_<span lang="fr-fr">chien chaud</span>_"), inconsistency in word selection throughout paragraphs, and even completely wrong translations.